code/modules/mob/mob.dm

```diff
@@ -1773,5 +1777,48 @@ mob/proc/on_foot()
 	ear_deaf = init_deaf
 	clear_fullscreen("blind")
 
+/mob/proc/transmogrify(var/target_type, var/offer_revert_spell = FALSE)
+	if(!target_type)
+		if(transmogged_from)
+			transmogged_from.forceMove(loc)
+			if(key)
+				transmogged_from.key = key
+			transmogged_from.timestopped = 0
+			if(istype(transmogged_from, /mob/living/carbon))
+				var/mob/living/carbon/C = transmogged_from
+				if(C.wear_mask && istype(C.wear_mask, /obj/item/clothing/mask/morphing))
+					C.drop_item(C.wear_mask, force_drop = 1)
+			transmogged_from = null
+			for(var/atom/movable/AM in contents)
+				AM.forceMove(get_turf(src))
+			qdel(src)
+		return
+	var/mob/M = new target_type(loc)
+	if(!istype(M))
+		qdel(M)
+		return
+	M.transmogged_from = src
+	if(key)
+		M.key = key
+	if(offer_revert_spell)
+		var/spell/change_back = new /spell/aoe_turf/revert_form
+		M.add_spell(change_back)
+	src.forceMove(null)
+	timestopped = 1
+
+/spell/aoe_turf/revert_form
```

Comment:
I think you can use targeted spells and set the target to only yourself. Not elegant but it's something.